finagle-http2/src/test/scala/com/twitter/finagle/http2/ClientFailUpgradeTest.scala

```diff
@@ -25,7 +25,8 @@ class ClientFailUpgradeTest extends AbstractHttp1EndToEndTest {
     TooLongStream,
     Streaming,
     CloseStream,
-    StreamFixed
+    StreamFixed,
+    NoBodyMessage
```

Comment:
my suspicion is that this fails for ClientFailUpgradeTest because the client starts having issued an upgrade message, so it's expecting a 101 in response.  When it gets a 101, it expects upgrade headers, and netty will throw an exception if you don't have them.  I don't know how the finagle client behaves in that scenario.  Do the tests work if you remove the 101 test from ClientFailUpgradeTest?  Could we restructure the tests to use "featureImplemented" to decide whether to include 101 or not instead of just turning them off for CFUT?